provisioning/roles/php/tasks/debian-repo.yml

```diff
@@ -1,24 +1,38 @@
 - name: debian-repo | error when trying to use something less than PHP 5.6 on jessie
   fail: msg="Jessie only supports version 5.6."
-  when: "ansible_lsb.major_release|int >= 8 and {{ php_version_installed }} < 5.6"
+  when: "ansible_lsb.major_release|int >= 8 and {{ php_version_installed | version_compare('5.6', '<') }}"
 
-- name: debian-repo | Install DotDeb key
+- name: debian-repo | error when trying to use something less than PHP 7 on Stretch
+  fail: msg="Stretch only supports version 7"
+  when: "ansible_lsb.major_release|int >= 9 and {{ php_version_installed | version_compare('7.0', '<') }}"
+
+- name: debian-repo | Install DotDeb key on jessie
   apt_key: url=https://www.dotdeb.org/dotdeb.gpg
   become: yes
-  when: "{{ php_version_installed }} > 5.4"
+  when: "ansible_lsb.major_release|int <= 8 and {{ php_version_installed | version_compare('5.4', '>') }}"
 
 - name: debian-repo | Make sure APT supports HTTPS sources
   apt: pkg=apt-transport-https state=present
   become: yes
 
-- name: debian-repo | Install deb.sury.org repository key
+- name: debian-repo | Install packages.sury.org repository key on jessie/stretch
   apt_key: id=AC0E47584A7A714D url=https://packages.sury.org/php/apt.gpg state=present
-  when: "{{ php_version_installed| version_compare('7.0', '>=') }}"
+  when: "ansible_lsb.major_release|int >= 7 and {{ php_version_installed| version_compare('7.0', '>=') }}"
```

Comment:
Debian 7 is wheezy, but the task name states `jessie/stretch`. Please update either the task name or the version comparison.